src/csharp/Microsoft.Spark/Sql/PicklingUdfWrapper.cs

```diff
@@ -274,24 +390,52 @@ internal object Execute(int splitIndex, object[] input, int[] argOffsets)
     [UdfWrapper]
     internal class PicklingUdfWrapper<T1, T2, T3, T4, T5, T6, T7, T8, T9, TResult>
     {
+        [NonSerialized]
+        private bool? _sameT1 = null;
+        [NonSerialized]
+        private bool? _sameT2 = null;
+        [NonSerialized]
+        private bool? _sameT3 = null;
+        [NonSerialized]
+        private bool? _sameT4 = null;
+        [NonSerialized]
+        private bool? _sameT5 = null;
+        [NonSerialized]
+        private bool? _sameT6 = null;
+        [NonSerialized]
+        private bool? _sameT7 = null;
+        [NonSerialized]
+        private bool? _sameT8 = null;
+        [NonSerialized]
+        private bool? _sameT9 = null;
+
```

Comment:
Any reason for not using an array? I guess b/c it's not serialized (requires allocation)? Maybe we can just serialize and use array?

Reply:
changed to array.